lib/gui/app/components/finish/finish.tsx

```diff
@@ -41,8 +41,33 @@ function restart(goToMain: () => void) {
 
 function FinishPage({ goToMain }: { goToMain: () => void }) {
 	const [webviewShowing, setWebviewShowing] = React.useState(false);
-	const errors = flashState.getFlashResults().results?.errors;
-	const results = flashState.getFlashResults().results || {};
+	let errors = flashState.getFlashResults().results?.errors;
+	if (errors === undefined) {
+		errors = (store.getState().toJS().failedDevicePaths || []).map(
```

Comment:
The errors are the same, but those in the flash results are not available if validation is skipped. I'll change the variable name from `failedDevicePaths` to something more clear and use only that as a reference, since it's more reliable